src/init/init.c

```diff
@@ -6,11 +6,34 @@
 /*   By: uschmidt <uschmidt@student.42berlin.de>    +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2025/03/18 14:03:56 by uschmidt          #+#    #+#             */
-/*   Updated: 2025/03/25 15:52:47 by uschmidt         ###   ########.fr       */
+/*   Updated: 2025/04/10 15:18:58 by jhelbig          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "init.h"
+#include "../malloc_list/malloc_list.h"
+
+char	**copy_envp(char **envp, t_list **malloc_list)
+{
+	int		size;
+	int		i;
+	char	**copy;
+
+	malloc_list = NULL;
+	size = 0;
+	while (envp[size])
+		size++;
+	copy = (char **)malloc(sizeof(char *) * (size + 1));
+	i = 0;
+	while (envp[i])
+	{
+		copy[i] = ft_strdup(envp[i]);
+		//add_to_malloc_list(malloc_list, copy[i]);
```

Comment:
Looks like you interrupted the envp feature on this branch to continue on the feat_envp, leaving some rudiments here. I see two options to deal with this here. either you clean this branch from all related changes, or, probably the better way would be to first merge the envp feature branch in main and then main into this branch. But there will be some merge conflicts either way